File: daySix/Java/TestQuestionFour.java
import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.*;

public class TestQuestionFour {

    @Test
    void TestThatIfIEnterAnArrayOfIntegersIGetTheNumberOfDuplicateInElementsInTheArray(){
        //Given
        int [] arrayOfNumbers = {45,60,3,0,67,2,45,3,22,0};
        
        //When
        int actualCountOfDuplicateElements = QuestionFour.countDuplicateElementsIn(arrayOfNumbers);
        
        int expectedCountOfDuplicateElements = 3;
        
        //Check
        assertEquals(expectedCountOfDuplicateElements, actualCountOfDuplicateElements);
    }
    
    @Test
    void TestThatWhenIGetAnArrayOfIntegersIGetAnArrayOfTheElementsThatAreDuplicated(){
        //Given
        int [] arrayOfNumbers = {45,60,3,0,67,2,45,3,22,0};
        
        //When 
        int [] actualArrayOfDuplicatedElements = QuestionFour.getArrayOfDuplicateElement(arrayOfNumbers);
        
        int [] expectedArrayOfDuplicatedElements = {45,3,0};
        
        //Check
        assertArrayEquals(expectedArrayOfDuplicatedElements, actualArrayOfDuplicatedElements);
    }
}

File: daySix/python/test_question_four.py
from unittest import TestCase

from question_four import *

class TestQuesionFour (TestCase):

    def test_thatWhenIGet_anArrayOfIntegers_iGetAnArray_ofTheElements_thatAreDuplicated(self):
    
        array_of_numbers = [45,60,3,0,67,2,45,3,22,0]
        
        actualArrayOfDuplicatedElements = get_array_of_duplicate_element(array_of_numbers)
        
        expectedArrayOfDuplicatedElements = [45,3,0]
        
        self.assertEqual(expectedArrayOfDuplicatedElements, actualArrayOfDuplicatedElements)
